video/out/wayland_common.c

```diff
@@ -2716,6 +2783,18 @@ int vo_wayland_control(struct vo *vo, int *events, int request, void *arg)
     return VO_NOTIMPL;
 }
 
+void vo_wayland_handle_hdr_metadata(struct vo_wayland_state *wl)
+{
+    if (!wl->color_surface || !wl->opts->wl_colorspace_hint || !pl_color_space_is_hdr(&wl->vo->params->color))
```

Comment:
`wl->vo->target_params->color` is probably more correct. `params` is input video params, while what we want to set is the thing that is after renderer, hence the `target_params`. This might also fix the issue with saturation that you observed

Reply:
Ah yup. That fixed it.